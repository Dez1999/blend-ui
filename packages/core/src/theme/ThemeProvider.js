import React, { useContext } from "react";
import styled, {
  ThemeProvider as StyledThemeProvider,
  ThemeContext,
} from "styled-components";
import { default as defaultTheme, baseStyles } from "./theme";

/*
const GlobalStyle = createGlobalStyle`
body: {
  margin:0;
};
`;
*/

/*
export const Base = styled.div`
  font-family: ${props => props.theme.fonts.body || null};
  font-weight: ${props => props.theme.fontWeights.medium || null};
  line-height: ${props => props.theme.lineHeights.standard || null};

  * {
  box-sizing: border-box;
};
body: {
  margin:0;
};
p: {
  margin: 0;
};

  * {
    box-sizing: border-box;
  }
  p: {
    margin: 0;
  }
`;
*/

export const Base = styled.div`
  ${baseStyles}
`;
const ThemeProvider = ({ theme = {}, ...props }) => {
  //console.log("THEME PROVIDER ", theme, defaultTheme);
  //console.log(baseStyles);
  const mergedTheme = { ...defaultTheme, ...theme };
  return (
    <StyledThemeProvider theme={mergedTheme}>
      <Base {...props} />
    </StyledThemeProvider>
  );
};

ThemeProvider.defaultProps = {
  defaultTheme,
};

const useTheme = () => {
  const theme = useContext(ThemeContext);
  if (theme === undefined) {
    throw new Error("useTheme must be used within a ThemeProvider");
  }
  return theme;
};

export default ThemeProvider;
export { useTheme };
