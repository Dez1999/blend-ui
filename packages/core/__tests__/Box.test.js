import React, { createRef } from "react";
import renderer from "react-test-renderer";
import Box from "../src/Box";

import ThemeProvider from "../src/theme/ThemeProvider";

test("Box renders correctly", () => {
  const tree = renderer
    .create(
      <ThemeProvider>
        <Box
          width={[1 / 2]}
          height={"100px"}
          border={"1px solid"}
          borderRadius={4}
          p={2}
        >
          Something
        </Box>
      </ThemeProvider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("Box renders sth with Ref correctly", () => {
  const boxRef = createRef();
  const tree2 = renderer
    .create(
      <ThemeProvider>
        <Box
          width={[1 / 2]}
          height={"100px"}
          border={"1px solid"}
          borderRadius={4}
          p={2}
          ref={boxRef}
          foo={"bar"}
        >
          Something with Ref
        </Box>
      </ThemeProvider>,
    )
    .toJSON();
  expect(tree2).toMatchSnapshot();
});
