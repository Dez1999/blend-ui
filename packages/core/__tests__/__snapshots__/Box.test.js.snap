// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box renders correctly 1`] = `
<div
  className="ThemeProvider__Base-sc-1w52cfi-0 eyQfDY"
>
  <div
    className="Box__BoxElement-sc-12sd5m8-0 gIcLpI"
  >
    Something
  </div>
</div>
`;

exports[`Box renders sth with Ref correctly 1`] = `
<div
  className="ThemeProvider__Base-sc-1w52cfi-0 eyQfDY"
>
  <div
    className="Box__BoxElement-sc-12sd5m8-0 gIcLpI"
    foo="bar"
  >
    Something with Ref
  </div>
</div>
`;
